internal/config/config.go

```diff
@@ -0,0 +1,168 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	defaultVaultAddress                 string = "https://127.0.0.1:8200"
+	defaultKubernetesServiceAccountPath string = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	defaultVaultKubernetesMountPath     string = "kubernetes"
+)
+
+// Config represents all of the provider's configurable behaviour from the MountRequest proto message:
+// * `parameters` from the SecretProviderClass (serialised into the `Attributes` field in the proto).
+// * Plus the rest of the proto fields we consume.
+// See sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1/service.pb.go
+type Config struct {
+	Parameters
+	TargetPath     string
+	FilePermission os.FileMode
+}
+
+// Parameters stores the parameters specified in the SecretProviderClass.
+// Top-level values that aren't strings are not directly deserialisable because
+// they are defined as literal string types:
+// https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/0ba9810d41cc2dc336c68251d45ebac19f2e7f28/apis/v1alpha1/secretproviderclass_types.go#L59
+//
+// So we just deserialise by hand to avoid complexity and two passes.
+type Parameters struct {
+	VaultRoleName                string
+	VaultAddress                 string
+	VaultKubernetesMountPath     string
+	KubernetesServiceAccountPath string
+	TLSConfig                    TLSConfig
+	Secrets                      []Secret
+}
+
+type TLSConfig struct {
+	VaultCAPEM         string
+	VaultCACertPath    string
+	VaultCADirectory   string
+	VaultTLSServerName string
+	VaultSkipTLSVerify bool
+}
+
+func (c TLSConfig) CertificatesConfigured() bool {
```

Comment:
Future enhancement should also include a configurable to specify the SNI of the server you're connecting to.

Cert SANs can be hard to get right, so sometimes you need to override the CA verification process and say "if this SAN is present in the cert, then its valid". We have this config in Vault Agent and its very useful: https://www.vaultproject.io/docs/commands#vault_tls_server_name)

Reply:
SGTM, I'll add that in a future PR.